instrumentation/httpx/lib/opentelemetry/instrumentation/httpx/instrumentation.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module HTTPX
+      # The Instrumentation class contains logic to detect and install the Http instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        install do |_config|
+          require_dependencies
+          patch
+        end
+
+        present do
+          defined?(::HTTPX)
+        end
+
+        option :peer_service, default: nil, validate: :string
+
+        def patch
+          otel_session = ::HTTPX.plugin(Plugin)
+
+          ::HTTPX.send(:remove_const, :Session)
+          ::HTTPX.send(:const_set, :Session, otel_session.class)
```

Comment:
Would you be able to help me understand why this code is replacing the Session class?

Reply:
httpx ships with its own plugin system, where one can load functionality on an instance without affecting the others: `session_with_retries = HTTPX.plugin(:retries)`. However, in the case of OTel and other adapters, the behaviour needs to be loaded by default for all instances, so the default session class is overridden by default + otel plugin. This looks ugly, but it's a require-time pass.

Reply:
Thanks for the explanation. As long as you are comfortable with this approach I understand why this needs to be done this way. 

Do other tracing plugins do something similar? E.g. DataDog Adapter you shared in one of your other comments?

Reply:
yes, all the other plugins go about a similar approach 👍